TESTS/mbed_hal/sleep/main.cpp

```diff
@@ -105,6 +105,11 @@ void sleep_usticker_test()
 
     const ticker_irq_handler_type us_ticker_irq_handler_org = set_us_ticker_irq_handler(us_ticker_isr);
 
+    // call ticker_read_us to initialize ticker upper layer
```

Comment:
looking at the lines above, there's if #0 - should also be removed

Reply:
To remove this, we need to have fix for `MBED_ALL_STATS_ENABLED` 

@maciejbocianski Please provide details

Reply:
When `MBED_ALL_STATS_ENABLED` is enable then `sleep_manager_sleep_auto` function calls some extra logging code, what makes return from `sleep` takes more time than expected

Reply:
Could we just unset the flag instead?

Reply:
impossible to do, this flag is set globally for all tests

Jenkins code:
```mbed_test_build_cmd=(mbed test --compile -m ${target} -t ${toolchain} -DMBED_HEAP_STATS_ENABLED=1 -DMBED_STACK_STATS_ENABLED=1 -DMBED_TRAP_ERRORS_ENABLED=1 -DMBED_ALL_STATS_ENABLED)```